configure.ac

```diff
@@ -86,13 +88,16 @@ fi
 
 AC_MSG_CHECKING(for special libxml2 includes)
 if test "x$XML2CONFIG" = "x"; then
-   AC_MSG_ERROR(libxml2 config not found)
+   PKG_CHECK_MODULES(XML2, [libxml-2.0],,
+		     [AC_MSG_ERROR(libxml2 config not found)])
+   XML2HEAD="$(XML2_CFLAGS)"
 else
    XML2HEAD="`$XML2CONFIG --cflags`"
-   AC_MSG_RESULT($XML2HEAD)
-   AC_CHECK_LIB(xml2, xmlReadMemory)
 fi
 
+AC_MSG_RESULT($XML2HEAD)
+AC_CHECK_LIB(xml2, xmlReadMemory)
```

Comment:
It'd be wise to also substitute `xmlReadDoc` for `xmlReadMemory`, since
the latter is not used (has actually never been, see 88c3d6a).

Reply:
feel free to push to this branch. Let's collect all the fixes for the build system that are required to integrate in CI and minor cleanups.. there is a reason why git a shared collaborative tool :P

Reply:
NP, just stating things that should be raked before merging,
can do those changes as well, if there are no further objections
or counterproposals.  Collaborative way, as you say :)